[ao] Create framework for ModelReport Qconfig Generation

vspenubarthi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83091

Summary: This creates the framework in the ModelReport API for the
generation of QConfigs by the ModelReport instance based on suggestions.
This functionality will eventually be added into the report generation
or be something that complements it, but for now it will be an
independent call for API stability and to be able to better modularize
the features as it stabilizes.

This also adds the framework for the relavent test function and a note
in the README at what future changes are planned for this new method in
the ModelReport API.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_qconfig_generation

Reviewers:

Subscribers:

Tasks:

Tags:

vspenubarthi committed:
[ao] Create framework for ModelReport Qconfig Generation

Summary: This creates the framework in the ModelReport API for the
generation of QConfigs by the ModelReport instance based on suggestions.
This functionality will eventually be added into the report generation
or be something that complements it, but for now it will be an
independent call for API stability and to be able to better modularize
the features as it stabilizes.

This also adds the framework for the relavent test function and a note
in the README at what future changes are planned for this new method in
the ModelReport API.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_qconfig_generation

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented on torch/ao/quantization/fx/_model_report/model_report.py:
```diff
@@ -415,3 +419,21 @@ def generate_visualizer(self) -> ModelReportVisualizer:
 
         return visualizer
+
+    def generate_qconfigs(self) -> Dict[str, QConfigMapping]:
```
may be better to have an equalization generator and a qconfigmapping generator since this type of dict object of qconfigmappings isn't something that exists elsewhere.

HDCharles replied:
would be better to name this either generate_qconfig_mappings or generate_qconfig_mapping/generate_equalization_config i.e. the actual name of the objects, because a qconfig is something different from a qconfig mapping

HDCharles reviewed (approved):
otherwise looks good

vspenubarthi committed:
Update on "[ao] Create framework for ModelReport Qconfig Generation"


Summary: This creates the framework in the ModelReport API for the
generation of QConfigs by the ModelReport instance based on suggestions.
This functionality will eventually be added into the report generation
or be something that complements it, but for now it will be an
independent call for API stability and to be able to better modularize
the features as it stabilizes.

This also adds the framework for the relavent test function and a note
in the README at what future changes are planned for this new method in
the ModelReport API.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_qconfig_generation

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g

vspenubarthi committed:
Update on "[ao] Create framework for ModelReport Qconfig Generation"


Summary: This creates the framework in the ModelReport API for the
generation of QConfigs by the ModelReport instance based on suggestions.
This functionality will eventually be added into the report generation
or be something that complements it, but for now it will be an
independent call for API stability and to be able to better modularize
the features as it stabilizes.

This also adds the framework for the relavent test function and a note
in the README at what future changes are planned for this new method in
the ModelReport API.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_qconfig_generation

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g
